ale/transformation.py

```diff
@@ -1,194 +1,99 @@
 
-    def __init__(self, id, parent=None, rotation=None):
-        """
-        Construct a frame node with or without a parent. If a parent is specified
-        then a rotation from this frame to the parent node's frame must be
-        specified and visa-versa.
-
-        Parameters
-        ----------
-        id : int
-             The NAIF ID code for the frame
-        parent : FrameNode
-                 The parent node in the frame chain
-        rotation : ConstantRotation or TimeDependentRotation
-                   The rotation from this frame to the frame of the parent node
-        """
+        times = np.array(ephemeris_time)
+        quats = np.zeros((len(times), 4))
 
-        if (parent is None) != (rotation is None):
-            raise TypeError("parent and rotation must both be entered or both be None.")
-
-        self.children = []
-        self.id = id
-        if parent is not None:
-            self.parent = parent
-        if rotation is not None:
-            self.rotation = rotation
-
-    def __repr__(self):
-        return f'ID:{self.id}\nChildren:{self.children}'
-
-    def __del__(self):
-        """
-        Custom deletor for a FrameNode. The child node is always responsible
-        for updating the parent nodes. So this removes this node from the
-        children of its parent node.
-        """
-        if self.parent is not None:
-            self.parent.children.remove(self)
-
-    @property
-    def parent(self):
-        """
-        The parent node of this node. Returns None if this is a root node.
-        """
-        if hasattr(self, '_parent'):
-            return self._parent
-        else:
-            return None
-
-    @parent.setter
-    def parent(self, new_parent):
-        """
-        Sets a new parent node. The child node is always responsible for
-        updating the parent node. So, this removes this node from the children
-        of the old parent and adds it to the children of the new parent.
-        """
-        if self.parent is not None:
-            self._parent.children.remove(self)
-
-        new_parent.children.append(self)
-        self._parent = new_parent
-
-    def parent_nodes(self):
-        """
-        Returns the ordered list of parents starting with this node going to
-        the root node.
-        """
-        chain = [self]
-        current_parent = self.parent
-        while current_parent is not None:
-            chain.append(current_parent)
-            current_parent = current_parent.parent
-        return chain
+        for s, d in frame_changes:
+            for i, time in enumerate(times):
+                rotation_matrix = spice.pxform(spice.frmnam(s), spice.frmnam(d), time)
+                quat_from_rotation = spice.m2q(rotation_matrix)
+                quats[i,:3] = quat_from_rotation[1:]
+                quats[i,3] = quat_from_rotation[0]
+            rotation = TimeDependentRotation(quats, times, s, d)
```

Comment:
Not all of the rotations in the frame chain are going to be time dependent rotations. That said, the time dependent vs constant rotation is kind of a headache right now. I could see just going to all time dependent.

Reply:
I'm wrestling with this right now. Looks like ISIS has a good chunk of logic that gathers the various frames from spice to generate the time dependent frame sets and the constant frame sets. Not sure the best way to go about solving this currently

Reply:
For right now, we are ignoring that logic because the methods for that are not wrapped in spiceypy yet.

Reply:
The only time dependent rotations we have to deal with are those from spiceinit'd ISIS cubes.

Reply:
Fair, this does result in problems regarding to_isis and to_usgscsm will be different depending on the source. If you do to_usgscsm from a dawn isis cube you'll get an additional constant rotation that you would have if you use to_usgscsm on a dawn pds image.

Reply:
to_usgscsm doesn't care about constant vs time dependent rotations, everything boils down to 1 time dependent rotation anyway. This is a problem for to_isis and it will need to be adressed under ALE work.